### PR TITLE
Refactor the pay controller

### PR DESCRIPTION
**Pull request by darkmoves**

### CODE REVIEW

File: app/controllers/pay_controller.rb [hunk 1]
```diff
@@ -4,43 +4,26 @@
...
-	  	  # update the rego record
-	  	  event = SpecialEvent.find_by url_code: url_code
-	  	  rego = EventRegistration.find_by payment_email: payer, special_event_id: event.id, id: rego_id.to_i
-	  	  rego.paid = true
-	  	  rego.save
-	  end
-	
-	  render :text => params
-	end
+	  txn = Hash["payer" => payer, "rego_id" => rego_id, "txn_id" => txn_id, "amount" => amount, "status" => pp_status, "event" => url_code]
```
Comment (darkmoves) [inline app/controllers/pay_controller.rb]: @brendanc what is `pp_status` here? I couldn't find it defined anywhere and need a value in order to write a test for the `#catch` method.

**Comment by darkmoves** [comment]
The spec for the `#catch` method isn't currently passing so this PR isn't quite ready to be merged. I'd also like to write a more comprehensive test for it.

File: app/controllers/pay_controller.rb (hunk 1 above)
Comment (brendanc) [inline app/controllers/pay_controller.rb]: ppstatus is payment_status from the IPN, see docs:  https://developer.paypal.com/webapps/developer/docs/classic/ipn/integration-guide/IPNandPDTVariables/

Right now we only really handle `Completed` status

**Comment by darkmoves** [comment]
This PR is ready for review. I did a smoke test on the demo app to ensure that a new marathon registration was appropriately marked as paid. @brendanc, I'd appreciate your eyes on this when you get a chance.

**Comment by darkmoves** [comment]
Tested the flow on demo. Looks good.

**Pull request merged**